src/sql/workbench/services/objectExplorer/browser/asyncServerTreeRenderer.ts

```diff
@@ -92,7 +92,7 @@ class ConnectionProfileTemplate extends Disposable {
 		super();
 		container.parentElement!.classList.add('connection-profile');
 		this._root = dom.append(container, dom.$('.connection-tile'));
-		this._icon = dom.append(this._root, dom.$('div.icon server-page'));
+		this._icon = dom.append(this._root, dom.$('div.icon'));
```

Comment:
Does this not work if you just make the selector `div.icon.server-page`? renderServerIcon will clear the ID if no path is found so we shouldn't need to redo that logic here.

Reply:
this also works, I just don't like to always set the default icon and then overwrite it by the id selector because id selector has higher priority.

Reply:
I guess I will do it in a cleaner way, here we will create a div with no class names and in the set method if iconPath is available => clear the class names, otherwise clear the id and set class names: icon server-page. how about this? @Charles-Gagnon

Reply:
btw, renderServerIcon doesn't clear the ID right now.

![image](https://user-images.githubusercontent.com/13777222/96056739-3c741800-0e3c-11eb-8f90-fdf776368cc6.png)

Reply:
Yeah that works.

Whoops, missed that part. TBH that truthy iconPath check probably shouldn't be there - putIcon handles an undefined iconPath already. No big deal to leave it as is for the time being though - while it'd be nice to clean this stuff up now I plan on going through and doing a big cleanup once I finish the async server tree work anyways.